src/ResourceManager/DeviceProvisioningServices/Commands.DeviceProvisioningServices/Cmdlet/IotDpsLinkedHubs/AddAzureRmIotDeviceProvisioningServiceLinkedHub.cs

```diff
@@ -0,0 +1,191 @@
+            ValueFromPipeline = true,
+            HelpMessage = "IoT Device Provisioning Service Object")]
+        [ValidateNotNullOrEmpty]
+        public PSProvisioningServiceDescription DpsObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "IoT Device Provisioning Service Resource Id")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the Resource Group")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Name of the IoT Device Provisioning Service")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            HelpMessage = "Connection String of the Iot Hub resource.")]
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "Connection String of the Iot Hub resource.")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Connection String of the Iot Hub resource.")]
+        [ValidateNotNullOrEmpty]
+        public string IotHubConnectionString { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = InputObjectParameterSet,
+            HelpMessage = "Location of the Iot Hub")]
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ParameterSetName = ResourceIdParameterSet,
+            HelpMessage = "Location of the Iot Hub")]
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            ParameterSetName = ResourceParameterSet,
+            HelpMessage = "Location of the Iot Hub")]
+        [ValidateNotNullOrEmpty]
+        [LocationCompleter("Microsoft.Devices/ProvisioningServices")]
+        public string IotHubLocation { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Allocation weight of the IoT Hub")]
+        [ValidateNotNullOrEmpty]
+        public int? AllocationWeight { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Apply allocation policy to the IoT Hub")]
+        [ValidateNotNullOrEmpty]
```

Comment:
@anusapan nit: you can remove the `ValidateNotNullOrEmpty` attribute from this parameter

Reply:
Done